salt/_pillar/metalk8s.py

```diff
@@ -11,26 +11,58 @@
 DEFAULT_SERVICE_NETWORK = '10.96.0.0/12'
 
 
+def _assert_equals(source_dict, expected_dict):
+    error_tplt = "Expected value '{}' for key '{}', got '{}'"
+
+    errors = [
+        error_tplt.format(value, key, source_dict.get(key))
+        for key, value in expected_dict.items()
+        if source_dict.get(key) != value
+    ]
+
+    return errors
+
+
+def _assert_keys(source_dict, keys):
+    errors = []
+    errors_tplt = "Expected presence of key '{}' in data: {}"
+
+    for key in keys:
+        if key not in source_dict:
+            errors.append(errors_tplt.format(key, source_dict))
+
+    return errors
```

Comment:
Maybe make sense to add these functions to a custom `utils` as it's could be usefull for all ext_pillar
https://docs.saltstack.com/en/latest/topics/utils/index.html

Reply:
Yup, planned :slightly_smiling_face: